sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -228,14 +228,21 @@ object SharedState extends Logging {
       sparkConf: SparkConf,
       hadoopConf: Configuration,
       initialConfigs: scala.collection.Map[String, String] = Map.empty): Unit = {
+
+    def containsInSparkConf(key: String): Boolean = {
+      sparkConf.contains(key) || sparkConf.contains("spark.hadoop." + key) ||
```

Comment:
(This is not related to this PR though) The string prefix `"spark.hadoop."` seems to be used in many places, so how about defining a global value for it somewhere? I think that the refactoring could make it easy to track the positions where the prefix is used.